openfeature-provider/java/src/main/java/com/spotify/confidence/RemoteMaterializationStore.java

```diff
@@ -0,0 +1,276 @@
+    this.clientSecret = clientSecret;
+    this.stub = createAuthStub(channelFactory, clientSecret);
+    this.readTimeout = getReadTimeout();
+    this.writeTimeout = getWriteTimeout();
+  }
+
+  @VisibleForTesting
+  RemoteMaterializationStore(
+      InternalFlagLoggerServiceGrpc.InternalFlagLoggerServiceBlockingStub stub,
+      String clientSecret,
+      Duration readTimeout,
+      Duration writeTimeout) {
+    this.stub = stub;
+    this.clientSecret = clientSecret;
+    this.readTimeout = readTimeout;
+    this.writeTimeout = writeTimeout;
+  }
+
+  private static InternalFlagLoggerServiceGrpc.InternalFlagLoggerServiceBlockingStub createAuthStub(
+      ChannelFactory channelFactory, String clientSecret) {
+    ManagedChannel channel = createConfidenceChannel(channelFactory);
+    return InternalFlagLoggerServiceGrpc.newBlockingStub(channel)
+        .withInterceptors(
+            new ClientInterceptor() {
+              @Override
+              public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(
+                  MethodDescriptor<ReqT, RespT> method, CallOptions callOptions, Channel next) {
+                return new ForwardingClientCall.SimpleForwardingClientCall<ReqT, RespT>(
+                    next.newCall(method, callOptions)) {
+                  @Override
+                  public void start(Listener<RespT> responseListener, Metadata headers) {
+                    headers.put(
+                        Metadata.Key.of("authorization", Metadata.ASCII_STRING_MARSHALLER),
+                        "ClientSecret " + clientSecret);
+                    super.start(responseListener, headers);
+                  }
+                };
+              }
+            });
+  }
+
+  private static Duration getReadTimeout() {
+    String envVal = System.getenv("CONFIDENCE_MATERIALIZATION_READ_TIMEOUT_SECONDS");
+    if (envVal != null && !envVal.isEmpty()) {
+      try {
+        return Duration.ofSeconds(Long.parseLong(envVal));
+      } catch (NumberFormatException e) {
+        logger.warn(
+            "Invalid CONFIDENCE_MATERIALIZATION_READ_TIMEOUT_SECONDS value: {}, using default",
+            envVal);
+      }
+    }
+    return DEFAULT_READ_TIMEOUT;
+  }
+
+  private static Duration getWriteTimeout() {
+    String envVal = System.getenv("CONFIDENCE_MATERIALIZATION_WRITE_TIMEOUT_SECONDS");
+    if (envVal != null && !envVal.isEmpty()) {
+      try {
+        return Duration.ofSeconds(Long.parseLong(envVal));
+      } catch (NumberFormatException e) {
+        logger.warn(
+            "Invalid CONFIDENCE_MATERIALIZATION_WRITE_TIMEOUT_SECONDS value: {}, using default",
+            envVal);
+      }
+    }
+    return DEFAULT_WRITE_TIMEOUT;
+  }
+
+  @Override
+  public CompletionStage<List<ReadResult>> read(List<? extends ReadOp> ops) {
+    if (ops == null || ops.isEmpty()) {
+      return CompletableFuture.completedFuture(Collections.emptyList());
+    }
+
+    return CompletableFuture.supplyAsync(
```

Comment:
AI suggestion: let's use a dedicated Executor for our tasks, it's cleaner than resorting to the shared "ForkJoinPool"